### DISCUSSION TITLE
Schmid factor Visualizing issue

### DISCUSSION THREAD

**Discussion by King-wang-H** [mtex-toolbox/mtex#1110] [Ask Anything]
Dear MTEX user,
I have calculated the Schmid factor through MTEX, now I want to visualize the Schmid factor on the pole figure, how should I achieve it?The final image I want to make is similar to the one below.
[
![C0E%471N9FV{)}%836QICBN](https://user-images.githubusercontent.com/85743677/123515518-4be57980-d6ca-11eb-9210-61b853c2f1fd.png)
](url)
I am particularly unfamiliar with this area, and I would appreciate it if you could provide me with a demo code.
Thank you in advance to all MTEX users.

Sincerely,
Wang

**Comment by King-wang-H** [comment]
Can someone help me?

**Comment by zmichels** [comment]
Maybe something like the following can get you going or close to what you want? I’ve never really tried to make this type of plot before, but I think this is along the right lines maybe (?).

%% Define the relevant crystal symmetry:
 
% just as an example, we will use a hexagonal crystal symmetry:
cs = crystalSymmetry('hexagonal');
 
% basal plane defined in Miller indices:
h = Miller(0,0,0,1,cs);
 
 
%% Definition of basal slip system
sS = slipSystem.basal(cs).symmetrise;
 
 
%% Define a stress tensor to match some deformation (RD/TD/ND directions)
 
% for... 
% RD = xvector
% TD = yvector ...
 
sigma = stressTensor([0 0 0; 0 -1 0; -1 0 0]);
 
% visualize the stress tensor:
figure,
plot(sigma)
annotate([vector3d.X, vector3d.Y],'label',{'RD', 'TD'},'backgroundColor','w')
mtexColorMap(flipud(red2blueColorMap))
saveFigure('Stress_Tensor.png','-bestfit')
 
%% Create an equispaced S2 grid
r = equispacedS2Grid('resolution',2.5*degree);
 
%% define crystal orientations with [0001] aligned to S2 grid
o = orientation.map(h,r);
 
%% Slip-systems in specimen coordinates
sSLocal = o * sS;
 
%% Compute the Schmid factor:
SF = sSLocal.SchmidFactor(sigma);
 
% take the maximum allong the rows
[SFMax,~] = max(SF,[],2);
 
%% Construct a spherical function, with weigths from SFMax
S2F = calcDensity(o*h,'weights',SFMax,'antipodal','halfwidth',15*degree);
 
%% Make a plot
figure,
plot(S2F,'antipodal','smooth','figSize','medium')
annotate([vector3d.X, vector3d.Y],'label',{'RD', 'TD'},'backgroundColor','w')
mtexColorMap(WhiteJetColorMap)
saveFigure('SchmidFactor_polar.png','-bestfit')


Hopefully that helps give you some traction… probably someone else can suggest a better approach, eventually.

Cheers,
Zach

**Reply by zmichels** [reply]
![SchmidFactor_polar](https://user-images.githubusercontent.com/8769619/123588827-fa161f80-d7ad-11eb-88f3-c243a5ba7002.png)

**Reply by King-wang-H** [reply]
Thank you very much.Thank you , It's really been a great help for me.
